spec/integration/args_injection_spec.rb

```diff
@@ -9,7 +9,7 @@ module PassThroughInitializer
             attr_reader :module_var
 
             def initialize(*)
-              super
+              super()
```

Comment:
the original case was described in https://github.com/dry-rb/dry-auto_inject/pull/14#issuecomment-218562210 which was based on Hanami's code: https://github.com/hanami/controller/blob/858303b5cf2d158f0d34f37a58ad384cb4468e0f/lib/hanami/action/mime.rb#L172
It seems I can work around this using the args strategy

Reply:
@flash-gordon Ah, thanks for finding this. This is actually very interesting to revisit. That PR addressed the `args` strategy only, since at that point it was the default injection strategy. A couple of months later, with the 0.3.0 release (2016-06-02), the kwargs strategy became the default and the spec we committed in #14 continued to work purely because of internal implementation details of the method we generate in kwargs’ `#define_initialize_with_splat`. These details changed here in this PR, which is why this `args_injection_spec` broke.

I feel like we need to take a few steps here:

1. Restore the spec to what it was before this PR (using `super` instead of `super()`), but make it use the `args` strategy, as was the intention when the spec was written (`AutoInject = Dry::AutoInject({one: 1}).args`). I've tested this locally and it works, as you'd expect. This would be enough to get this current PR merged.
2. Add similar behaviour to the kwargs strategy, since it seems like it should be a helpful thing to incorporate in all our strategies, if possible.
3. Improve our descriptions of these specs so it's easier to understand their use cases simply by reading the spec files, now having to go back through PR history :)

What do you think? I'll commit the change for (1) right now, and if you're OK with this and all of my other changes today, we can get this PR merged.